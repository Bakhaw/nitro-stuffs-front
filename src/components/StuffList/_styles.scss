.StuffList {
  display: flex;
  flex: 1;
  flex-direction: row;
  margin: 30px 0;

  &__Filter {
    margin-left: 40px;
  }

  &__Stuffs {
    display: flex;
    flex: 10;
    flex-direction: column;

    &__List {
      display: grid;
      grid-template-columns: repeat(auto-fill, 350px);
      gap: 30px;
      justify-content: center;

      a {
        text-decoration: none;
      }
    }
  }
}
