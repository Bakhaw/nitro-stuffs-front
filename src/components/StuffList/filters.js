const filters = [
  {
    name: 'Smokes',
    type: 'smoke'
  },
  {
    name: 'Flashbangs',
    type: 'flashbang'
  },
  {
    name: 'Molotovs',
    type: 'molotov'
  }
];

export default filters;
