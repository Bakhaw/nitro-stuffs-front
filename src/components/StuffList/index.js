import React from 'react';
import { Link } from 'react-router-dom';

import StuffCard from '../StuffCard';

const stuffs = [
  {
    id: '9avd4ed',
    title: 'Connector from T Spawn',
    type: 'smoke',
    videoId: 'kua_hX-l-V4' // youtube video ID
  },
  {
    id: '9avd4ed',
    title: 'Connector from T Spawn',
    type: 'smoke',
    videoId: 'kua_hX-l-V4' // youtube video ID
  },
  {
    id: '9avd4ed',
    title: 'Connector from T Spawn',
    type: 'smoke',
    videoId: 'kua_hX-l-V4' // youtube video ID
  },
  {
    id: '9avd4ed',
    title: 'Connector from T Spawn',
    type: 'smoke',
    videoId: 'kua_hX-l-V4' // youtube video ID
  },
  {
    id: '9avd4ed',
    title: 'Connector from T Spawn',
    type: 'smoke',
    videoId: 'kua_hX-l-V4' // youtube video ID
  }
];

function StuffList() {
  return (
    <ul className='StuffList'>
      {stuffs.map(stuff => (
        <li key={stuff.id}>
          <Link to={`/game/csgo/stuff/${stuff.id}`}>
            <StuffCard stuff={stuff} />
          </Link>
        </li>
      ))}
    </ul>
  );
}

export default StuffList;
