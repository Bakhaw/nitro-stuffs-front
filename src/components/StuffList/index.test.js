import React from 'react';
import { shallow } from 'enzyme';

import StuffList from '.';

test('should render properly', () => {
  const props = {
    activeStuffs: [],
  };

  const container = shallow(<StuffList {...props} />);
  expect(container.debug()).toMatchSnapshot();
});
