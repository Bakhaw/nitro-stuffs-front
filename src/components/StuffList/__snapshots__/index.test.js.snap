// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<div className=\\"StuffList\\">
  <div className=\\"StuffList__Title\\">
    <h1>
      MIRAGE
    </h1>
  </div>
  <ul className=\\"StuffList__Stuffs\\" />
</div>"
`;
